BinaryEncoding.md

```diff
@@ -317,18 +317,20 @@ It is legal to have several entries with the same type.
 | Name | Opcode | Immediate | Description |
 | ---- | ---- | ---- | ---- |
 | `nop` | `0x00` | | no operation |
-| `block` | `0x01` | count = `varuint32` | a sequence of expressions, the last of which yields a value |
-| `loop` | `0x02` | count = `varuint32` | a block which can also form control flow loops |
-| `if` | `0x03` | | high-level one-armed if |
-| `if_else` | `0x04` | | high-level two-armed if |
+| `block` | `0x01` |  | begin a sequence of expressions, the last of which yields a value |
+| `loop` | `0x02` |  | begin a block which can also form control flow loops |
+| `if` | `0x03` | | begin if expression |
+| `else` | `0x04` | | begin else expression of if |
 | `select` | `0x05` | | select one of two values based on condition |
 | `br` | `0x06` | relative_depth = `varuint32` | break that targets a outer nested block |
 | `br_if` | `0x07` | relative_depth = `varuint32` | conditional break that targets a outer nested block |
 | `br_table` | `0x08` | see below | branch table control flow construct |
-| `return` | `0x14` | | return zero or one value from this function |
-| `unreachable` | `0x15` | | trap immediately |
+| `return` | `0x09` | | return zero or one value from this function |
+| `unreachable` | `0x0a` | | trap immediately |
+| `end` | `0x0f` | | end a block, loop, or if |
```

Comment:
@titzer I see it differently. We already have the case of zero-values for `nop` making it clear that it is not a values stack. If there were multiple value expressions and blocks passed through multiple values then each stack entry would represent an AST node and thus an expression and thus the results of that expression which could have multiple values or no values.

I think it would be subjectively nice for the following to be equivalent `(block (call $fn_returning_two_values)) == (call $fn_returning_two_values)` without having to do the following:

```
(block $l
  (destructuring_set_local (10 11) (call $fn_returning_two_values))
  (br_with_atomic_tuple $l (get_local 10) (get_local 11)))
```

Also to be able to write `(block ... (tuple (get_local 10) (get_local 11))` rather than having to use an atomic `br_with_atomic_tuple`.

Also for the following to be equivalent  `(call $fn_returning_two_values) == (block (call $fn_returning_two_values)) == (block $l (br $l (call $fn_returning_two_values)))`.